sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/DefaultPodSpec.java

```diff
@@ -28,6 +28,8 @@
     @NotNull
     @Min(0)
     private final Integer count;
+    @NotNull
+    private Boolean allowDecommission;
```

Comment:
Does this still default to `false` if we add the `@NotNull` annotation?

Reply:
(do we ever instantiate this class through something other than a builder?)

Reply:
These classes are serialized into ZK, I think this gets triggered during deserialization from ZK.

Reply:
From a quick peek at the code, we run the validation either in the builder or in the constructor that deserialization would use (or both). Should be a-ok.

Reply:
The only instantiation paths are via builders or the `@JsonCreator` constructor. And the builders use the latter.